File: 이건희/level2/연속된 부분 수열의 합.py
def solution(sequence, k):
    result = []
    sumNum = 0
    now = 0
    s = 0

    while now < len(sequence):
        sumNum += sequence[now]
        
        if sumNum > k:
            while sumNum > k:
                sumNum -= sequence[s]
                s += 1
        
        if sumNum == k:
            if len(result) == 0 or now - s < result[1] - result[0]:
                result = [s, now]
                
        now += 1
        
    return result                

File: 이훈일/Programmers/Level2/연속된 부분 수열의 합.java
class Solution {
    public int[] solution(int[] sequence, int k) {
        int[] answer = new int[]{0, sequence.length-1};
        int left = 0;
        int right = 0;
        int sum = sequence[0];
        while(true) {
            if(sum < k && right < sequence.length-1) {
                right++;
                sum += sequence[right];
            } else if(sum > k) {
                sum -= sequence[left];
                left++;
            }
            
            if(sum == k){
                if(right-left < answer[1]-answer[0]) {
                    answer[0] = left;
                    answer[1] = right;
                } else if(right-left == answer[1]-answer[0]) {
                    if(left < answer[0]) {
                        answer[0] = left;
                        answer[1] = right;
                    }
                }
                
                if(right < sequence.length-1) {
                    right++;
                    sum += sequence[right];
                }
            }
            
            if(right == sequence.length-1) {
                if(sum <= k) {
                    break;
                }
            }
        }
        
        return answer;
    }
}
